fake-snippets-api/routes/api/snippets/remove_star.ts

```diff
@@ -0,0 +1,43 @@
+import { accountSnippetSchema } from "fake-snippets-api/lib/db/schema"
+import { withRouteSpec } from "fake-snippets-api/lib/middleware/with-winter-spec"
+import { z } from "zod"
+
+export default withRouteSpec({
+  methods: ["POST"],
+  auth: "session",
+  jsonBody: z.object({
+    snippet_id: z.string(),
+  }),
+  jsonResponse: z.object({
+    ok: z.boolean(),
+    account_snippet: accountSnippetSchema,
+  }),
+})(async (req, ctx) => {
+  const { snippet_id } = req.jsonBody
+
+  // Check if snippet exists
+  const snippet = ctx.db.getSnippetById(snippet_id)
+  if (!snippet) {
+    return ctx.error(404, {
+      error_code: "snippet_not_found",
+      message: "Snippet not found",
+    })
+  }
+
+  // Remove star
+  ctx.db.removeStar(ctx.auth.account_id, snippet_id)
+
+  // parse account snippet
+  const accountSnippet = {
+    account_id: ctx.auth.account_id,
+    snippet_id,
+    has_starred: false,
+    created_at: new Date().toISOString(),
+    updated_at: new Date().toISOString(),
+  }
+
+  return ctx.json({
+    ok: true,
+    account_snippet: accountSnippetSchema.parse(accountSnippet),
```

Comment:
```suggestion
```

Reply:
should i just return `snippet.is_starred` with an ok response here?

Reply:
The snippet from `/snippets/get` should have `snippet.is_starred` based on the user

Reply:
when you star it, `qc.invalidQueries` is used to make `/snippets/get` repull the latest version of the snippet including the star status